internal/ir/ir.go

```diff
@@ -352,6 +352,20 @@ type HashMap struct {
 	Value Value
 }
 
+// SubscriptExpr is an IR Value that represents the access of an element of an array.
+//
+// Example printed form:
+// %t0 = array["a","b","c"]
+// %t1 = "0"
+// %t2 = %t0[%t1]
+//
+// The SubscriptExpr implements the Value interface.
```

Comment:
Since the subscription can compute a a new value (return the value being acceded), this struct should also implements the `Instruction` interface.